webui/react/src/components/kit/Select.tsx

```diff
@@ -0,0 +1,163 @@
+import { Select as AntdSelect } from 'antd';
+import type { DefaultOptionType, LabeledValue, RefSelectProps, SelectValue } from 'antd/es/select';
+import React, { forwardRef, useCallback, useMemo, useState } from 'react';
+
+import Label, { LabelTypes } from 'components/Label';
+import Icon from 'shared/components/Icon/Icon';
+
+import css from './Select.module.scss';
+
+const { OptGroup, Option } = AntdSelect;
+
+export { Option, SelectValue };
+
+type Options = DefaultOptionType | DefaultOptionType[];
+export interface Props<T = SelectValue> {
+  allowClear?: boolean;
+  defaultValue?: T;
+  disableTags?: boolean;
+  disabled?: boolean;
+  enableSearchFilter?: boolean;
+  filterOption?: boolean | ((inputValue: string, option: LabeledValue | undefined) => boolean);
+  filterSort?: (a: LabeledValue, b: LabeledValue) => 1 | -1;
+  id?: string;
+  label?: string;
+  maxTagCount?: number | 'responsive';
+  maxTagPlaceholder?: string;
+  mode?: 'multiple' | 'tags';
+  onBlur?: () => void;
+  onChange?: (value: T, option: Options) => void;
+  onDeselect?: (selected: SelectValue, option: Options) => void;
+  onSearch?: (searchInput: string) => void;
+  onSelect?: (selected: SelectValue, option: Options) => void;
+  options?: LabeledValue[];
+  placeholder?: string;
+  placement?: 'bottomLeft' | 'bottomRight' | 'topLeft' | 'topRight';
+  ref?: React.Ref<RefSelectProps>;
+  value?: T;
+}
+
+const countOptions = (children: React.ReactNode, options?: Options): number => {
```

Comment:
bug: found a bug when `Option` child components are used in `Select` component.
`option` component accepts `ReactNode` type, so lots of elements can be nested inside `option`. That calculates the `countOption` incorrectly. 
The quick fix i can think of is that we can stop accepting `children` and always let developers to use `options={[]}` props instead of `option` children. It is aso much easier to calculate the number of children. That also makes code simpler. What do you think?

You can reproduce the bug with the following code

```tsx
<Select mode="multiple">
  <Option value={'all'}>All Projects</Option>
  <Option value={'Projects'}>My Projects</Option>
  <Option value={'other Projects'}>
    <div>Others&apos; Projects</div>
  </Option>
  <Option value={'other Projects'}>
    <div>Others&apos; Projects</div>
  </Option>
</Select>
```

Reply:
@keita-determined the value in your example is never getting set. Try this: 

```
  const [exampleValues, setExampleValues] = useState<SelectValue>();

<Select value={exampleValues} onChange={(value) => setExampleValues(value)} disableTags mode="multiple">
  <Option key="all" value={'all'}>All Projects</Option>
  <Option key="Projects" value={'Projects'}>My Projects</Option>
  <Option key="other Projects1" value={'other Projects'}>
    <div>Others&apos; Projects</div>
  </Option>
  <Option key="other Projects" value={'other Projects again'}>
    <div>Others&apos; Projects</div>
  </Option>
</Select>
```

Reply:
https://user-images.githubusercontent.com/103522725/222574924-33171ca1-d9a1-4a3e-9f73-59652809efcd.mov

Reply:
Also let me know if I am wrong somewhere and am misunderstanding the bug you are mentioning.

Reply:
i misunderstood this recursion. seems like its working.